pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1349,8 +1347,14 @@ protected void handleSeek(CommandSeek seek) {
             Consumer consumer = consumerFuture.getNow(null);
             Subscription subscription = consumer.getSubscription();
             MessageIdData msgIdData = seek.getMessageId();
+            BitSetRecyclable ackSet = BitSetRecyclable.create();
+            if (msgIdData.hasBatchIndex()) {
+                ackSet.set(0, Math.max(msgIdData.getBatchIndex(), 0));
+            }
 
-            Position position = new PositionImpl(msgIdData.getLedgerId(), msgIdData.getEntryId());
+            Position position = new PositionImpl(msgIdData.getLedgerId(),
+                    msgIdData.getEntryId(), ackSet.toLongArray());
+            ackSet.recycle();
```

Comment:
It's better tot check `msgIdData.hasBatchIndex()` first, If the conditions are not met, we don't need to create the BitSet instance.